addons/website/models/ir_module_module.py

```diff
@@ -500,14 +500,15 @@ def _load_module_terms(self, modules, langs, overwrite=False):
         for generic_arch_db, specific_arch_db, specific_id in self.env.cr.fetchall():
             if not generic_arch_db:
                 continue
-            langs_update = (langs & generic_arch_db.keys()) - {'en_US'}
-            specific_langs = (langs & specific_arch_db.keys()) - {'en_US'}
-            generic_arch_db_en = generic_arch_db.pop('en_US')
-            specific_arch_db_en = specific_arch_db.pop('en_US')
-            generic_arch_db = {k: generic_arch_db[k] for k in langs_update}
-            specific_arch_db = {k: specific_arch_db.get(k, specific_arch_db_en) for k in specific_langs}
-            generic_translation_dictionary = field.get_translation_dictionary(generic_arch_db_en, generic_arch_db)
-            specific_translation_dictionary = field.get_translation_dictionary(specific_arch_db_en, specific_arch_db)
+            langs_update = (langs & generic_arch_db.keys() & specific_arch_db.keys()) - {'en_US'}
```

Comment:
if the
langs is `nl_NL`
`generic_arch_db` has `en_US`, `fr_FR`, `nl_NL`
`specific_arch_db` has `en_US`, `fr_FR`
`langs_update` will become empty set
But I think you still want to use the new imported generic `nl_NL` translations to update `specific_arch_db` `nl_NL` right?

Reply:
This is still an open question for me... I have prepared a question for a colleague that is off today.

Reply:
@HydrionBurst To be sure in stable, I now add all translations as you suggested. Whether or not limiting it to less translations is a safe optimization can be found out later, in a future PR.